### PR TITLE
INTDEV-438 Validation: Do not accept extra files or folders in project directory

### PR DESCRIPTION
**Pull request by Fuzzbender**
Make validation more strict in regards to extra folders and files. 

Generally, no extra files or folders are allowed in project files. Exceptions are
- on root directory, there can be extra files, but no directories
- under each `a` (attachment) directory there can be any files

### CODE REVIEW

**Comment by korva** [comment]
Running `cyberismo validate` for current cyberismo-isms repo main branch I get:
`At '.calc' non-allowed additional property: resources`

**Comment by Fuzzbender** [comment]
> Running `cyberismo validate` for current cyberismo-isms repo main branch I get: `At '.calc' non-allowed additional property: resources`
> 
> And in unified-sdl: `At '.calc' non-allowed additional property: cardtree.lp`

.calc folder is generated when calculations are done. You get rid of old name of the lp-file nag by removing the folder and re-calculating and then validating. Not sure what the `resources` is, wasn't there previously. Probably something that needs to be just skipped (allowed).

**Comment by Fuzzbender** [comment]
I can't get `resources` to appear under `.calc`. 
Have you executed calculations with a version of binary that is not in main and it has updated the calculations?

**Comment by Fuzzbender** [comment]
I'll change the implementation so that it won't validate `.cards` folder at all. That folder can be seen as temp file for calculations and as such is ignored in validation.

**Comment by samutoljamo-pareto** [comment]
Yes, resources folder is made in #140 . I noticed, it would be easy to generate the other files(workflows, cardTypes) etc if we store them by their name. However, since the name is in the format project/type/name,  I didn't want to store them under .calc directly. I ended up putting all of those to /resources folder.

File: tools/schema/cardtree-directory-schema.json [hunk 1]
```diff
@@ -6,11 +6,18 @@
   "properties": {
     "directories": {
       "type": "object",
+      "additionalProperties": false,
```
Comment (samutoljamo-pareto) [inline tools/schema/cardtree-directory-schema.json:9]: Shouldn't we allow any properties at the root level?

Comment (Fuzzbender) [reply tools/schema/cardtree-directory-schema.json:9]: I think we could. I'll change it.

**Review by henryhaverinen — APPROVED** [review]

**Pull request merged**